### PR TITLE
docs: Adding Oscilloscope guide document

### PR DESCRIPTION
**Pull request by Vikum94**
First steps for issues #898 

Changes:

* Created the Oscilloscope.md document and added the main topics.

Screenshots for the change: None
>None

APK for testing: 
> None

### CODE REVIEW

File: docs/Oscilloscope.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+<table>
+    <tr>
+        <td><img src="/docs/screenshots/oscilloscope_channel_params.png"></td>
+        <td><img src="/docs/screenshots/oscilloscope_data_analysis.png"></td>
+    </tr>
+</table>
+<table>
+    <tr>
+        <td><img src="/docs/screenshots/oscilloscope_timebase_and_trigger.png"></td>
+        <td><img src="/docs/screenshots/oscilloscope_xy_plot.png"></td>
```
Comment (harsh-2711) [inline docs/Oscilloscope.md]: @Vikum94  Why have you put this image of app crashing?

Comment (Vikum94) [reply docs/Oscilloscope.md]: That was a mistake. Now it's updated :+1:

**Comment by CloudyPadmal** [comment]
@Vikum94 how is progress with this PR?

File: docs/Oscilloscope.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+### Observing external waves
+
+Any of the channels must be connected to the positive pin of the signal generator which you want to analyze. Use any of the ground pins in PSLab device to connect the negative/ground pin of the signal generator.
+
+### Observing waves from the in-built Wavegenerator instrument  
+
+Using the Wavegenerator, set values to Sine, Square or Trangular waves. Then the relevant pin you used should be connected to any of the channels. 
+For example, Let's generate two waves, a Sine wave and a Sqaure wave. Set the frequency using Wavegenerator by selecting Sine wave 1 and Square wave 1. Additionally you can change the phase and duty cycle of Square wave. Then connect the Channel 1 and Channel 2 to the Sine wave and Square wave as shown below.
+
+<img src="/docs/images/oscilloscope_schematic.png">
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: This image shows a connection of SI pin and SQ pin at the same time. PSLab does not support simultaneous square and sine wave generation

File: docs/Oscilloscope.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Open Oscilloscope and in the Channel parameters section, set the check boxes CH1 and CH2. You can observe the waves plotting. 
+Using Timebase, observe the plot changes when time range is increased.   
+
+## Video Demo
+
+<a href="http://www.youtube.com/watch?feature=player_embedded&v=Ua9_OCR4p8Y
+" target="_blank"><img src="http://img.youtube.com/vi/Ua9_OCR4p8Y/0.jpg"
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: This link is using old version of PSLab

**Comment by Vikum94** [comment]
@CloudyPadmal Changes done. Please have a look. The folder is renamed screenshots ==> images. I updated the readme file too. You might have seen it the changed files.

File: docs/Oscilloscope.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+2. Layout
+3. Features and Components
+    - Channel Parameters
+    - Timebase and Trigger
+    - Data analysis
+    - XY Plot
+4. How to Use  
+
+## Introduction
+Pocket Science Lab contains an Oscilloscpe which has lot of functions which are available in a commercial grade Oscilloscope. The PSLab android application communicates with the Pocket Science Lab and gives the user a real experience on using a commercial Oscilloscope. This document will help a user on available functionalities, how to use Pocket Science Lab with PSLab android application and some demo videos.
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: Pocket Science Lab contains an **_Oscilloscpe_** which --> Pocket Science Lab contains an **_Oscilloscope_** which

File: docs/Oscilloscope.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+
+Component | Name in the device
+--- | ---
+Channels | CH1, CH2, CH3, MIC
+Sine waves | SI1, SI2
+PWM Generators | SQ1, SQ2, SQ3, SQ4
+Ground | GND
+
+These features can be used using four components in the Oscilloscope. Note that GND is common for all the pins in the device.
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: I'm not sure how we can relate `SI1` `SI2` `SQ` pins with oscilloscope as they are features in waveform generator.

File: docs/Oscilloscope.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+### 1. Channel Parameters
+Channel parameters can be used to select the following settings.
+- Channel/s that needs to be observerd
+- Voltage range to be played in the plot
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: Add `Amplitude` term here

File: docs/Oscilloscope.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+These features can be used using four components in the Oscilloscope. Note that GND is common for all the pins in the device.
+
+### 1. Channel Parameters
+Channel parameters can be used to select the following settings.
+- Channel/s that needs to be observerd
+- Voltage range to be played in the plot 
+
+When the tick boxes selected from the items below the graph plot, user will be able to find the live plotting of the input read through the relevant channel. 
+For Channel 1 and Channel 2, user can adjust the Voltage range of the graphs 'y axis'. 
+The MIC can be used to read either internal Microphone or External microphone.
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: "internal Microphone or External microphone" all in simple is better right?

File: docs/Oscilloscope.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+Timebase parameter is used to change the range of 'x axis' which is Time. 
+Trigger takes two parameters channel and voltage. When the read voltage of the selected channel exceeds the selected value for voltage, a trigger occurs.
+
+### 3. Data Analysis
+This function is to fit the read data series into,
+1. Sine function
+2. Square function
+3. Fourier transform
+
+When the oscilloscope reads an input, using data analysis function, user can find out the closest sine/square function for the observed data input.
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: Can we make all `Oscilloscope` terms to `oscilloscope` with a simple `o`?

File: docs/Oscilloscope.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+### Observing waves from the in-built Wavegenerator instrument  
+
+Using the Wavegenerator, set values to Sine, Square or Trangular waves. Then the relevant pin you used should be connected to any of the channels. 
+For example, Let's generate a Sine wave. Set the frequency using Wavegenerator by selecting Sine wave 1. Additionally you can change the phase of the wave. Then connect the Channel 1 to the Sine wave as shown below.
+
+<img src="/docs/images/oscilloscope_schematic.png">
+
+Open Oscilloscope and in the Channel parameters section, set the check boxes CH1. You can observe the waves plotting. 
+Using Timebase, observe the plot changes when time range is increased. Similarly set Sine wave 2 frequency and phase and connect the SI2 pin to CH2. Check tick box CH2 in the Channel parameters. Select XY plot and tick the Enable XY plot checkbox. Observe the channel to channel voltage plot.
```
Comment (CloudyPadmal) [inline docs/Oscilloscope.md]: You can use some markdown syntax to emphasize label names from general text. Like `Enable XY plot`

**Pull request merged**